lib/game_logic.rb

```diff
@@ -0,0 +1,166 @@
+# rubocop: disable Style/ClassVars, Metrics/LineLength, Style/GuardClause
+require 'colorize'
+class UserInterface
+  def self.players
+    @@players
+  end
+
+  def initialize(name1, name2)
+    @name1 = name1
+    @name2 = name2
+    @@players = []
+    @@players << @name1
+    @@players << @name2
+  end
+
+  def self.show_players
+    "you are #{@@players[0]} your mark is (X) and you choose go vs #{@@players[1]} his mark is (O)"
+  end
+
+  def self.show_game_instructions
+    "Welcome! \n This is our Tic-Tac-Toc Game \n For this game you will need THE AMAZING INSTRUCTIONS: \n The game start in a grid table of 3x3 cells with numbers.\n You will be 2 players, the player 1 play first, and the second... play second, one turn by one, choosing what number play \n You need complete a line (horizontal, vertical or diagonal) with 3 grid-cells or numbers\n".green
+  end
+
+  def self.show_who_play
+    LogicGame.who_play?
+  end
+
+  def self.choose
+    @turn = LogicGame.turn
+    "you choose #{@turn}".blue
+  end
+
+  def self.invalid_move
+    'You choose an unvalid option, please try again and select an option from the available as shown'.red
+  end
+
+  def self.show_win_game
+    if !LogicGame.player_winner.nil?
+      LogicGame.player_winner
```

Comment:
- [ ] `[OPTIONAL]` you can use 

```Ruby
if LogicGame.player_winner
      LogicGame.player_winner
```
this means if `LogicGame.player_winner` is not false or nil, it will run.